apps/browser-extension-wallet/src/stores/slices/wallet-activities-slice.ts

```diff
@@ -53,7 +53,14 @@ const DelagatioinTransactionTypes = new Set(['delegation', 'delegationRegistrati
 
 const getDelegationAmount = (activity: AssetActivityItemProps) => {
   const fee = new BigNumber(Number.parseFloat(activity.fee));
-  return activity?.deposit ? fee.plus(activity.deposit) : fee;
+  return (
+    fee
+      .plus(activity.deposit || 0)
+      .minus(activity.returnedDeposit || 0)
+      // Currently, all values in activity screen are absolute
+      // Remove .abs() when this is changed
+      .abs()
+  );
```

Comment:
Can we be more explicit in this case? It requires a deep understanding of the flow to know that both `deposit` and `returnedDeposit` cannot be present at the same time.

After applying the above comment we can do the following:

```suggestion
  if (activity.type === 'delegationRegistration') {
    return fee.plus(activity.deposit);
  }
  
  if (activity.type === 'delegationDeregistration') {
    return new BigNumber(activity.returnedDeposit).minus(fee);
  }
  
  return fee;
```

Reply:
I hope that when you narrow the types you will mitigate unhandled edge cases + the QA phase will reveal if something is there. Technically it'd be more correct to do it this way, as we described but given the time pressure and that your logic also "does the job" 😄 it's for you to decide.